### ISSUE TITLE
Formula check is not working, and the goroutine  is not closed

### ISSUE THREAD

**Issue by ZhangYi96** [unidoc/unioffice#436]
## Description
I want to check and calculate the formula。

## Expected Behavior
When I input formula such as`ADD(1=1),1=1`, there will be an error in excel and I wana get an error.
## Actual Behavior
The program should have returned an error, but it didn't. It passed the vertification.
In the same time, the goroutine for LexReader can not be closed. It will write String `ADD(1=1),1=1` into chan node, but only String `ADD(1=1)` will be read, so the goroutine run gopark().
![image](https://user-images.githubusercontent.com/30099113/93553770-79b4c980-f9a6-11ea-8ae8-661bdacb14dc.png)

Please include a reproducible code snippet or document attachment that
demonstrates the issue.

**Comment by zgordan-vv** [comment]
@ZhangYi96 Hi, I can;t reproduce the error. When I write something like ```sheet.Cell("A4").SetFormulaRaw("ADD(1=1),1=1")
    a4Value := formEv.Eval(sheet.FormulaContext(), "A4")``` UniOffice gives me `parse error: syntax error: unexpected tokenComma`. And when I am trying to imput it in Excel, it doesn't let me input it at all. Can you tell me exact steps to reproduce the issue?

**Comment by ZhangYi96** [comment]
I am using the v0.7.1 version, but I also found this problem in the latest version.
Yes, it can check the formula, but it print the error not return

**Test**
```
func TestEval(t *testing.T) {
	td := []struct {
		Inp string
		Exp string
	}{
		{"AND(1=1),1=1", "1 ResultTypeNumber"},
	}

	ss := spreadsheet.New()
	sheet := ss.AddSheet()
	ctx := sheet.FormulaContext()

	ev := formula.NewEvaluator()
	for _, tc := range td {
		t.Run(tc.Inp, func(t *testing.T) {
			p := formula.Parse(strings.NewReader(tc.Inp))
			if p == nil {
				t.Errorf("error parsing %s", tc.Inp)
				return
			}
			result := p.Eval(ctx, ev)
			got := fmt.Sprintf("%s %s", result.Value(), result.Type)
			if got != tc.Exp {
				t.Errorf("expected %s = %s, got %s", tc.Inp, tc.Exp, got)
			}
		})
	}
}

```

**OutPut**
```
=== RUN   TestEval
=== RUN   TestEval/AND(1=1),1=1
2020/10/09 11:52:08 parse error: syntax error: unexpected tokenComma
--- PASS: TestEval (0.00s)
    --- PASS: TestEval/AND(1=1),1=1 (0.00s)
PASS
```
it print the error, but pass the test. In web services, when similar formulas are calculated (the parentheses end early), the lex goroutine will not exit. My program once had 1 million goroutines blocked at lex.

In addition, The formula is not read from excel, but obtained through the input box on the webpage, so I cannot guarantee the correctness of the formula before calculating.

Thank you for your reply

**Comment by gunnsth** [comment]
This has been addressed in v1.7.1. Multiple improvements in formula parsing and evaluation including timeouts and such to avoid possible go routine leaks and crashes.
https://github.com/unidoc/unioffice/releases/tag/v1.7.1
Please test to confirm that it resolves the issue.
